pkg/skaffold/runner/runner.go

```diff
@@ -166,24 +167,69 @@ func (r *SkaffoldRunner) dev(artifacts []*config.Artifact) error {
 	return nil
 }
 
-func (r *SkaffoldRunner) run(artifacts []*config.Artifact) (*build.BuildResult, *deploy.Result, error) {
+func (r *SkaffoldRunner) buildAndDeploy(artifacts []*config.Artifact) (*build.BuildResult, *deploy.Result, error) {
+	bRes, err := r.build(artifacts)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "build step")
+	}
+
+	r.builds = mergeWithPreviousBuilds(bRes.Builds, r.builds)
+
+	dRes, err := r.deploy(&build.BuildResult{
+		Builds: r.builds,
+	})
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "build step")
+	}
+
+	return bRes, dRes, nil
+}
+
+func (r *SkaffoldRunner) build(artifacts []*config.Artifact) (*build.BuildResult, error) {
 	start := time.Now()
 	fmt.Fprintln(r.opts.Output, "Starting build...")
+
 	bRes, err := r.Builder.Build(r.opts.Output, r.Tagger, artifacts)
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "build step")
+		return nil, errors.Wrap(err, "build step")
 	}
+
 	fmt.Fprintln(r.opts.Output, "Build complete in", time.Now().Sub(start))
 
-	start = time.Now()
+	return bRes, nil
+}
+
+func (r *SkaffoldRunner) deploy(bRes *build.BuildResult) (*deploy.Result, error) {
+	start := time.Now()
 	fmt.Fprintln(r.opts.Output, "Starting deploy...")
-	if _, err := r.Deployer.Deploy(r.opts.Output, bRes); err != nil {
-		return nil, nil, errors.Wrap(err, "deploy step")
+
+	dRes, err := r.Deployer.Deploy(r.opts.Output, bRes)
+	if err != nil {
+		return nil, errors.Wrap(err, "deploy step")
 	}
 	if r.opts.Notification {
 		fmt.Fprint(r.opts.Output, constants.TerminalBell)
 	}
+
 	fmt.Fprintln(r.opts.Output, "Deploy complete in", time.Now().Sub(start))
 
-	return bRes, nil, nil
+	return dRes, nil
+}
+
+func mergeWithPreviousBuilds(builds, previous []build.Build) []build.Build {
+	updatedBuilds := map[string]bool{}
```

Comment:
any issue with just putting map[string]build.Build on SkaffoldRunner instead of the slice?

Reply:
no, no issue. Do you think it would be easier/clearer?

Reply:
Nope, might be slightly cleaner, but shouldn't really make much of a difference.